### PR TITLE
Modifying Rook Installation & Adding Storage Monitoring Components

### PR DESCRIPTION
**Pull request by umangachapagain**
For the Storage Console and Prometheus based Monitoring to work on top of OCP, there are few restrictions that needs to be addressed. This PR addresses those and adds Storage Monitoring feature to the stack.

Changes: 
1. Rook + Ceph cluster is installed in the openshift-storage namespace.
2. Service Monitor is added to openshift-storage namespace.
3. Prometheus Rule file is applied from the ceph-mixins project to enable alerting.

Signed-off-by: Umanga Chapagain <chapagainumanga@gmail.com>

### CODE REVIEW

File: 10_deploy_rook.sh [hunk 1]
```diff
@@ -64,3 +79,11 @@ spec:
...
     activeStandby: true
 EOF
+
+cd $ROOKPATH/cluster/examples/kubernetes/ceph/monitoring
+sed 's/namespace: rook-ceph/namespace: openshift-storage/' service-monitor.yaml > service-monitor-modified.yaml
+sed -i 's/- rook-ceph/- openshift-storage/' service-monitor-modified.yaml
+oc create -f service-monitor-modified.yaml
+
+cd $MIXINPATH/extras/manifests
+oc create -f prometheus-rules.yaml
```
Comment (shtripat) [inline 10_deploy_rook.sh]: The rule file name is going to be same for dev script as well?

Comment (umangachapagain) [reply 10_deploy_rook.sh]: Whatever we decide. Currently, there are no other files that I can apply.

File: 10_deploy_rook.sh [hunk 2]
```diff
@@ -7,27 +7,46 @@ source common.sh
...
+oc policy add-role-to-user view system:serviceaccount:openshift-monitoring:prometheus-k8s -n openshift-storage
+
 sed '/FLEXVOLUME_DIR_PATH/!b;n;c\          value: "\/etc/kubernetes\/kubelet-plugins\/volume\/exec"' operator-openshift.yaml > operator-openshift-modified.yaml
 sed -i 's/# - name: FLEXVOLUME_DIR_PATH/- name: FLEXVOLUME_DIR_PATH/' operator-openshift-modified.yaml
+sed -i 's/namespace: rook-ceph/namespace: openshift-storage/' operator-openshift-modified.yaml
+sed -i 's/:rook-ceph:/:openshift-storage:/' operator-openshift-modified.yaml
 oc create -f operator-openshift-modified.yaml
 sleep 120
+
 oc wait --for condition=ready  pod -l app=rook-ceph-operator -n rook-ceph --timeout=120s
```
Comment (karmab) [inline 10_deploy_rook.sh]: those lines should be removed

Comment (umangachapagain) [reply 10_deploy_rook.sh]: ack.

**Comment by russellb** [comment]
would like to see a successful test run by one other person, and then we can merge.

**Comment by hardys** [comment]
@karmab Hey is this something you have tested?

File: 10_deploy_rook.sh [hunk 3]
```diff
@@ -7,27 +7,42 @@ source common.sh
...
 cd $ROOKPATH/cluster/examples/kubernetes/ceph
-oc create -f common.yaml
+
+sed 's/name: rook-ceph$/name: openshift-storage/' common.yaml > common-modified.yaml
+sed -i 's/namespace: rook-ceph/namespace: openshift-storage/' common-modified.yaml
+oc create -f common-modified.yaml
+oc label namespace openshift-storage  "openshift.io/cluster-monitoring=true"
+oc policy add-role-to-user view system:serviceaccount:openshift-monitoring:prometheus-k8s -n openshift-storage
+
 sed '/FLEXVOLUME_DIR_PATH/!b;n;c\          value: "\/etc/kubernetes\/kubelet-plugins\/volume\/exec"' operator-openshift.yaml > operator-openshift-modified.yaml
```
Comment (karmab) [inline 10_deploy_rook.sh]: no longer needed as it was merged upstream

Comment (umangachapagain) [reply 10_deploy_rook.sh]: ack.

**Comment by umangachapagain** [comment]
@karmab Fixed it. 
Also bumped ceph version to v14 as it is a requirement for Storage Components on Web UI.

File: 10_deploy_rook.sh [hunk 4]
```diff
@@ -7,27 +7,43 @@ source common.sh
...
 eval "$(go env)"
 
+CEPH_VERSION=v14
+export MIXINPATH="$GOPATH/src/github.com/ceph/ceph-mixins"
 export ROOKPATH="$GOPATH/src/github.com/rook/rook"
 cd $ROOKPATH/cluster/examples/kubernetes/ceph
-oc create -f common.yaml
-sed '/FLEXVOLUME_DIR_PATH/!b;n;c\          value: "\/etc/kubernetes\/kubelet-plugins\/volume\/exec"' operator-openshift.yaml > operator-openshift-modified.yaml
-sed -i 's/# - name: FLEXVOLUME_DIR_PATH/- name: FLEXVOLUME_DIR_PATH/' operator-openshift-modified.yaml
+
```
Comment (karmab) [inline 10_deploy_rook.sh]: we still need oc create -f common.yaml ;)

Comment (umangachapagain) [reply 10_deploy_rook.sh]: There's common-modified.yaml around line 17.

**Comment by karmab** [comment]
tested it without the image bumping and it worked, so merging

**Pull request merged**